ow_lander/scripts/activity_grind.py

```diff
@@ -27,32 +54,59 @@ def plan_cartesian_path_lin(move_arm, length):
 
 def grind(move_arm, move_limbs, args):
 
-  x_start = args[1]
-  y_start = args[2]
+  move_arm.set_planner_id('RRTstar')
+
+  x_start = 0.9*args[1]
+  y_start = 0.9*args[2] 
   depth = args[3]
   length = args[4]
-  ground_position = args[5]
+  radial = args[5]
+  ground_position = args[6]
 
-  pre_move_complete = move_to_pre_trench_configuration(move_arm, x_start, y_start)
-  if pre_move_complete == False:
-    return False
+  # Compute shoulder yaw angle to trench
+  alpha = math.atan2(y_start-constants.Y_SHOU, x_start-constants.X_SHOU)
+  h = math.sqrt( pow(y_start-constants.Y_SHOU,2) + pow(x_start-constants.X_SHOU,2) )
+  l = constants.Y_SHOU - constants.HAND_Y_OFFSET
+  beta = math.asin (l/h)
+  # Move to pre trench position, align shoulder yaw
+  joint_goal = move_arm.get_current_joint_values()
+  joint_goal[constants.J_DIST_PITCH] = math.pi/6
+  joint_goal[constants.J_HAND_YAW] = 4*math.pi/3
+  joint_goal[constants.J_PROX_PITCH] = -math.pi/2
+  joint_goal[constants.J_SHOU_PITCH] = math.pi/3
+  joint_goal[constants.J_SHOU_YAW] = alpha + beta
+  alpha = alpha+beta-0.09
 
-  # rotate hand
-  change_joint_value(move_arm, constants.J_HAND_YAW, -2*math.pi/3)
 
-  # approaching and entering terrain - along -Z
+  # If out of joint range, abort
+  if (is_shou_yaw_goal_in_range(joint_goal) == False):
+    return False
+
+  move_arm.go(joint_goal, wait=True)
+  move_arm.stop()
 
+  # entering terrain
   z_start = ground_position + constants.GRINDER_HEIGHT - depth
   go_to_Z_coordinate(move_limbs, x_start, y_start, z_start)
 
-  # grinding ice - along +X
-  cartesian_plan, fraction = plan_cartesian_path_lin(move_arm, length)
-  move_limbs.execute(cartesian_plan, wait=True)
-  move_limbs.stop()
+  # grinding ice forward
+  cartesian_plan, fraction = plan_cartesian_path(move_arm, length, alpha, radial)
+  move_arm.execute(cartesian_plan, wait=True)
+  move_arm.stop()
+
+  joint_goal = move_arm.get_current_joint_values()
+  change_joint_value(move_arm, constants.J_SHOU_YAW, joint_goal[0]+0.08)
+
+  # grinding ice backwards
+  cartesian_plan, fraction = plan_cartesian_path(move_arm, -length, alpha, radial)
+  move_arm.execute(cartesian_plan, wait=True)
+  move_arm.stop()
+
 
-  # exiting terrain - along +Z
-  z_start = ground_position + constants.GRINDER_OFFSET - depth
-  go_to_Z_coordinate(move_arm, x_start, y_start, z_start)
+  # exiting terrain
+  joint_goal = move_arm.get_current_joint_values()
+  change_joint_value(move_arm, constants.J_SHOU_PITCH, joint_goal[1]+0.6)
 
+  move_arm.set_planner_id('RRTconnect')
```

Comment:
Because otherwise the planner will keep using RRTstar for the entire session, until the sim is killed. However, this line is no longer required in the new commit.